runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/PackageUtil.java

```diff
@@ -397,10 +397,21 @@ public static PackageAttributes forFileToStage(
             String.format("Non-existent file to stage: %s", file.getAbsolutePath()));
       }
       checkState(!file.isDirectory(), "Source file must not be a directory.");
+      String target;
+      switch (dest) {
+        case "dataflow-worker.jar":
```

Comment:
I'm not sure why we need special treatment for Dataflow jars here. Behavior for these jars should be similar for any other jars uploaded for Dataflow IMO.

@lukecwik WDYT ?

Reply:
Can we add a log statement describing why we need special treatment for these jars ? I see these jars being specifically being referenced to in Dataflow boot.go files.May be that's why ?

Reply:
Put some comments and logging